lib/structs/audit_log_entry.ex

```diff
@@ -0,0 +1,123 @@
+defmodule Crux.Structs.AuditLogEntry do
+  @moduledoc """
+    Represents a Discord [Audit Log Object](https://discordapp.com/developers/docs/resources/audit-log)
+  """
+
+  @behaviour Crux.Structs
+
+  alias Crux.Structs
+  alias Crux.Structs.{AuditLogChange, Util}
+
+  @audit_log_events %{
+    guild_update: 1,
+    channel_create: 10,
+    channel_update: 11,
+    channel_delete: 12,
+    channel_overwrite_create: 13,
+    channel_overwrite_update: 14,
+    channel_overwrite_delete: 15,
+    member_kick: 20,
+    member_prune: 21,
+    member_ban_add: 22,
+    member_ban_remove: 23,
+    member_update: 24,
+    member_role_update: 25,
+    role_create: 30,
+    role_update: 31,
+    role_delete: 32,
+    invite_create: 40,
+    invite_update: 41,
+    invite_delete: 42,
+    webhook_create: 50,
+    webhook_update: 51,
+    webhook_delete: 52,
+    emoji_create: 60,
+    emoji_update: 61,
+    emoji_delete: 62,
+    message_delete: 72
+  }
+
+  @type event_name ::
+          :guild_update
+          | :channel_create
+          | :channel_update
+          | :channel_delete
+          | :channel_overwrite_create
+          | :channel_overwrite_update
+          | :channel_overwrite_delete
+          | :member_kick
+          | :member_prune
+          | :member_ban_add
+          | :member_ban_remove
+          | :member_update
+          | :member_role_update
+          | :role_create
+          | :role_update
+          | :role_delete
+          | :invite_create
+          | :invite_update
+          | :invite_delete
+          | :webhook_create
+          | :webhook_update
+          | :webhook_delete
+          | :emoji_create
+          | :emoji_update
+          | :emoji_delete
+          | :message_delete
+
+  @audit_log_events_key Map.new(@audit_log_events, fn {k, v} -> {v, k} end)
+
+  @doc """
+    Returns a map of all audit log event names with their id
+  """
+  @event_names Map.keys(@audit_log_events)
+  @spec events() :: %{event_name => non_neg_integer()}
+  def events, do: @audit_log_events
+
+  @doc """
+    Gets the event name from the action type id
+  """
+  @spec event_name(action_type :: non_neg_integer()) :: atom()
+  def event_name(action_type), do: Map.get(@audit_log_events_key, action_type)
+
+  defstruct(
+    id: nil,
+    target_id: nil,
+    changes: nil,
+    user_id: nil,
+    action_type: nil,
+    options: nil,
+    reason: nil
+  )
+
+  @type t :: %__MODULE__{
+          id: integer(),
+          target_id: integer(),
+          changes: [AuditLogChange.t()] | nil,
+          user_id: integer(),
+          action_type: integer(),
+          options: AuditLogOptionalEntry.t() | nil,
```

Comment:
Woops, I was debating whether or not I should make a struct where every field is optional... But since the docs don't title it as an Object like the rest of the structs, I think I'll make it just a map.